### DISCUSSION TITLE
definePageMeta ignored by nuxt-link

### DISCUSSION THREAD

**Discussion by Lucky-Strike1695** [nuxt/nuxt#17009] [Questions] [answered]
**Env :**
Platform : WSL - Debian
Nodejs : 16.17.0
Nuxt : 3.0.0.rc8

**What I'm trying to do :** 
Create a homemade replacement for nuxt-auth module.
An `auth.global.ts` middleware is called on every route. I need to somehow get a parameter to know if the targeted page is protected, unprotected, or 'guest' (like the login page, only accessible for _non_-logged in users).

**My issue :** 
The middleware is called without issue. But I was using `definePageMeta()` to send the type of route (protected/unprotected/guest) like so :

```javascript 
// pages/user/index.vue
definePageMeta({
  auth: true      // false, 'guest'
})
```
That meta is only accessible in middleware _IF_ I manually target the URL (going directly to 'https://domain/user/' or refreshing the page from this URL.
If I navigate from another page, using `nuxt-link` to reach /user, the meta is not caught and to the middleware does not prevent the access to the protected page.

```javascript
// middleware/auth.global.ts
// ...
const route = useRoute()
console.log(route.meta.auth) // undefined
```

Actually, it seems quite logical since that meta is on the targeted page and not the one calling the middleware. How to get metas from the _targeted_ page ?
If I should not be using `definePageMeta()`, what should I be using instead to achieve what I'm trying to do ?

Thanks a lot.

**Comment by danielroe** [comment, marked as answer]
You have access within middleware to two parameters - `to` and `from`. You can access the metadata on the `to` parameter to get whjat you want...

```js
export default defineNuxtRouteMiddleware(to => {
  // console.log(to.meta.auth)
})
```

**Reply by Lucky-Strike1695** [reply]
Brilliant. Will try asap !
Besides, I take from the absence of comment on the way I plan to do things that it isn't too stupid ;).
